subgraphs/burrow/src/helpers/protocol.ts

```diff
@@ -1,119 +1,161 @@
-import { dataSource, near, BigInt } from '@graphprotocol/graph-ts';
-import { LendingProtocol, UsageMetricsDailySnapshot, UsageMetricsHourlySnapshot, FinancialsDailySnapshot } from '../../generated/schema';
-import { BD_ZERO } from '../utils/const';
+import { dataSource, near, BigInt } from "@graphprotocol/graph-ts";
+import {
+  LendingProtocol,
+  UsageMetricsDailySnapshot,
+  UsageMetricsHourlySnapshot,
+  FinancialsDailySnapshot,
+} from "../../generated/schema";
+import {
+  BD_ZERO,
+  LendingType,
+  NANOSEC_TO_SEC,
+  Network,
+  ProtocolType,
+  PROTOCOL_NAME,
+  PROTOCOL_SLUG,
+  RiskType,
+  NANOS_TO_DAY,
+  NANOS_TO_HOUR,
+} from "../utils/const";
+import { Versions } from "../versions";
 
 export function getOrCreateProtocol(): LendingProtocol {
-	let protocol = LendingProtocol.load(dataSource.address().toString());
-	if (!protocol) {
-		protocol = new LendingProtocol(dataSource.address().toString());
-		protocol.name = 'Burrow';
-		protocol.slug = 'burrow';
-		protocol.schemaVersion = '2.0.1';
-		protocol.subgraphVersion = '1.0.0';
-		protocol.methodologyVersion = '1.0.0';
-		protocol.network = 'NEAR_MAINNET';
-		protocol.type = 'LENDING';
-		protocol.lendingType = 'POOLED';
-		protocol.riskType = 'GLOBAL';
-		protocol.totalValueLockedUSD = BD_ZERO;
-		protocol.cumulativeUniqueUsers = 0;
-		protocol.cumulativeSupplySideRevenueUSD = BD_ZERO;
-		protocol.cumulativeProtocolSideRevenueUSD = BD_ZERO;
-		protocol.cumulativeTotalRevenueUSD = BD_ZERO;
-		protocol.totalPoolCount = 0;
-		protocol._marketIds = new Array<string>();
-		protocol.save();
-	}
-	return protocol;
+  let protocol = LendingProtocol.load(dataSource.address().toString());
+  if (!protocol) {
+    protocol = new LendingProtocol(dataSource.address().toString());
+    protocol.name = PROTOCOL_NAME;
+    protocol.slug = PROTOCOL_SLUG;
+    protocol.schemaVersion = Versions.getSchemaVersion();
```

Comment:
these 3 lines should be on the outside of the if block

Reply:
If we ever in the future graft this subgraph with changes made it will not update the versions unless you place them outside of the if block.

I understand it may not be a feature in NEAR subgraphs, but it is a standard and improves our ability to maintain the subgraph over time